tests/clar_libgit2.c

```diff
@@ -319,6 +319,35 @@ const char* cl_git_path_url(const char *path)
 	return url;
 }
 
+const char *cl_git_sandbox_path(int is_dir, const char *path, ...)
+{
+	static char _temp[4096];
+	git_buf buf = GIT_BUF_INIT;
+	va_list arg;
+
+	cl_git_pass(git_buf_sets(&buf, clar_sandbox_path()));
+	if (path) {
+		cl_git_pass(git_buf_joinpath(&buf, buf.ptr, path));
+
+		va_start(arg, path);
+
+		while ((path = va_arg(arg, const char *)) != NULL) {
+			cl_git_pass(git_buf_joinpath(&buf, buf.ptr, path));
+		}
+		va_end(arg);
+	}
+
+	cl_git_pass(git_path_prettify(&buf, buf.ptr, NULL));
+	if (is_dir)
+		git_path_to_dir(&buf);
+
+	git_buf_copy_cstr(_temp, 4096, &buf);
```

Comment:
You should check whether this truncates `buf`, first. Agreed, it is very unlikely, but I don't want to clean up (rm -rf) truncated paths at any point in time.

Reply:
Fixed by checking the length of buf before copying.